### PR TITLE
Drop log4j2 by default

### PR DESCRIPTION
**Pull request by dixel**
Switch to using slf4j on the library level and an attempt to use timbre/timbre-slf4j on the template level.
This way the responsibility for handling logging now goes to the developers and interop between different logging backends becomes a bit easier.

### CODE REVIEW

File: template/resources/leiningen/new/friboo/core.clj [hunk 1]
```diff
@@ -2,14 +2,28 @@
...
+  ([data] (default-log-output-fn nil data))
+  ([_ data] (let [{:keys [level ?ns-str ?msg-fmt vargs ?err]} data]
+              (format "%5s [%s] %s - %s%s\n"
+                      (str/upper-case (name level))
+                      (.getName (Thread/currentThread))
+                      (str ?ns-str)
+                      (if-let [fmt ?msg-fmt]
+                        (apply format fmt vargs)
+                        (apply str vargs))
+                      (str (when ?err (log/stacktrace ?err)))))))
```
Comment (dixel) [inline template/resources/leiningen/new/friboo/core.clj:25]: Definitely had some hesitations about the way of configuring logging output here, tried to keep the format look like it was in the main library before and make the template work out of the box.

Comment (dryewo) [reply template/resources/leiningen/new/friboo/core.clj:25]: That's fine, we can make shortened namespaces later.

File: template/resources/leiningen/new/friboo/project.clj [hunk 2]
```diff
@@ -5,7 +5,9 @@
             :url  "http://www.apache.org/licenses/LICENSE-2.0"}
   :min-lein-version "2.0.0"
   :dependencies [[org.clojure/clojure "1.8.0"]
-                 [org.zalando.stups/friboo "2.0.0-beta5"]]
+                 [com.fzakaria/slf4j-timbre "0.3.4"]
+                 [com.taoensso/timbre "4.8.0"]
+                 [org.zalando.stups/friboo "2.0.0-beta7-SNAPSHOT"]]
```
Comment (dryewo) [inline template/resources/leiningen/new/friboo/project.clj]: I'd rather not make the template depend on a SNAPSHOT verison.

**Comment by dryewo** [comment]
👍

File: template/resources/leiningen/new/friboo/core.clj [hunk 3]
```diff
@@ -2,14 +2,28 @@
...
             [{{namespace}}.api :as api])
   (:gen-class))
 
 (def default-http-config
   {:http-port 8080})
 
+(defn default-log-output-fn
+  ([data] (default-log-output-fn nil data))
+  ([_ data] (let [{:keys [level ?ns-str ?msg-fmt vargs ?err]} data]
+              (format "%5s [%s] %s - %s%s\n"
```
Comment (dryewo) [inline template/resources/leiningen/new/friboo/core.clj]: \n is not needed here, it adds a blank line

**Comment by dryewo** [comment]
👍

**Comment by dixel** [comment]
:+1:

**Comment by dixel** [comment]
:+1:

**Comment by dixel** [comment]
:+1:

**Comment by dixel** [comment]
:+1:

**Comment by dixel** [comment]
:+1:

**Pull request merged**